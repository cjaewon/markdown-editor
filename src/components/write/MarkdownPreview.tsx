import * as React from 'react';
import styled from 'styled-components';
import MarkdownRender from '../common/MarkdownRender';
import palette from '../../lib/styles/palette';

const MarkdownPreviewBlock = styled.div`
  word-break: break-word;
  padding: 3rem;
  flex: 1;
  overflow-y: auto;
  color: ${palette.gray9};
`;

const Title = styled.h1`
  font-size: 2.5em;
  margin-bottom: 4rem;
  font-weight: 800;
`;

export interface MarkdownPreviewProps {
  markdown: string;
  title: string;
  className?: string;
}

const MarkdownPreview: React.FC<MarkdownPreviewProps> = ({
  markdown,
  title,
  className,
}) => {
  return (
    <MarkdownPreviewBlock id="preview">
      <Title>{title}</Title>
      <MarkdownRender markdown={markdown} editing />
    </MarkdownPreviewBlock>
  );
};

export default MarkdownPreview;
