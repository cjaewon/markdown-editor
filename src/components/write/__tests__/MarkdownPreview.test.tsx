import * as React from 'react';
import { render } from '@testing-library/react';
import MarkdownPreview, { MarkdownPreviewProps } from '../MarkdownPreview';

describe('MarkdownPreview', () => {
  const setup = (props: Partial<MarkdownPreviewProps> = {}) => {
    const initialProps: MarkdownPreviewProps = {
      markdown: '',
      title: '',
    };
    const utils = render(<MarkdownPreview {...initialProps} {...props} />);
    return {
      ...utils,
    };
  };
  it('renders properly', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('renders markdown', () => {
    const utils = setup({
      markdown: '## hello world',
    });
    const element = utils.getByText('hello world');
    expect(element.tagName).toBe('H2');
  });
  it('renders title', () => {
    const utils = setup({
      title: 'this is the title',
    });
    utils.getByText('this is the title');
  });
});
