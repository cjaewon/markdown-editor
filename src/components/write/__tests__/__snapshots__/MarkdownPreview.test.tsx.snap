// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownPreview matches snapshot 1`] = `
<div>
  <div
    class="sc-htpNat ixBInR"
    id="preview"
  >
    <h1
      class="sc-bxivhb cFXJHm"
    />
    <div
      class="sc-bdVaJa epxEIa"
    >
      <div
        class="sc-bwzfXH kcygLR atom-one-light"
      >
        

      </div>
    </div>
  </div>
</div>
`;
